daemon/algod/api/server/v2/dryrun.go

```diff
@@ -397,16 +397,19 @@ func doDryrunRequest(dr *DryrunRequest, response *generated.DryrunResponse) {
 		return
 	}
 	proto := config.Consensus[protocol.ConsensusVersion(dr.ProtocolVersion)]
-
+	proto.EnableAppCostPooling = true
 	response.Txns = make([]generated.DryrunTxnResult, len(dr.Txns))
 	for ti, stxn := range dr.Txns {
 		pse := logic.MakePastSideEffects(len(dr.Txns))
+		maxBudget := uint64(proto.MaxAppProgramCost * proto.MaxTxGroupSize)
+		evalBudget := maxBudget
```

Comment:
Maybe I miss something but this does not look correct. `maxBudget` is  MaxAppProgramCost * (num app call txns)

Reply:
Related to the previous idea: We're willing to be overly generous with the cost, so that the execution will first "succeed" but we could (with a bit extra code), properly turn the success to failure after the fact.  This would be bad in production (we're letting people use the CPU too much), but nice for dry-run.

That requires a bit of code that isn't there yet (I think, I haven't looked). It would determine the *real* maxCost as well, and confirm that the cost is under that.  But it would do it *after* execution, rather than setting the pooled cost exactly right from the start.

Reply:
then it should be heavily commented, otherwise it is not obvious at all